Sources/PackageGraph/ModulesGraph+Loading.swift

```diff
@@ -300,7 +316,13 @@ private func checkAllDependenciesAreUsed(
             if dependency.products.contains(where: \.isCommandPlugin) {
                 continue
             }
-            
+
+            // Skip this check if traits are enabled since it is valid to add a dependency just
+            // to enable traits on it.
```

Comment:
Can you elaborate on this in the comment? Unsure of an exact use case of enabling traits by adding a dependency. It would be great to have an example listed right there to clarify this point, otherwise it doesn't seem obvious enough.

Reply:
Updated the comment with more details